lib/sidebar.rb

```diff
@@ -3,211 +3,131 @@
 
-        child = "<li class=\"#{item_class}\">"
-        child << "<div class=\"nav-group-heading\"><i class=\"material-icons\">arrow_right</i><span>#{value[:title].split('–').first}</span></div>"
+      def write_sidebar_to_files(files)
+        files.each do |file|
+          sanitized_filename = file[:sanitized_filename]
+          filename = file[:filename]
+          doc = file[:doc]
+          sidebar_markup = render(sanitized_filename, pages)
 
-        child << '<ul class="nav-ul">'
+          next if sidebar_markup.nil?
 
-        subgroup = generateSubgroup(filename, key, value, subgroups, 2)
+          sidebar_containers = find_containers(doc, filename)
 
-        child << subgroup
+          next if sidebar_containers.nil?
 
-        if subgroups.any?
-          subgroups.select { |subgroup_key, _subgroup_value| subgroup_key.split('/').length <= 3 }.each do |subgroup_key, subgroup_value|
-            subgroup = generateSubgroup(filename, subgroup_key, subgroup_value, subgroups, 2)
-            child << subgroup
+          sidebar_containers.each do |sidebar_container|
+            sidebar_container.inner_html = sidebar_markup
           end
-        end
-
-        child << '</ul>'
-        child << '</li>'
-        sidebar << child
-      end
-
-      File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
-      sidebar
-    end
 
-    def active?(filename, url, exact = false)
-      if filename == '/' || url == '/'
-        if filename == '/' && url == '/'
-          return true
-        else
-          return false
+          Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+          File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
         end
       end
 
-      exact ? filename == url : filename.start_with?(url)
-    end
-
-    def merge(hash1, hash2)
-      all_keys = hash1.keys | hash2.keys
-      result_hash = {}
+      def render(filename, pages)
+        sidebar_markup = nil
 
-      all_keys.each do |key|
-        hash_value = {}
+        begin
+          builder = SidebarBuilder.new(filename, pages)
 
-        if hash1.key? key
-          value = hash1[key]
+          sidebar_markup = builder.build
 
-          hash_value = value unless value.nil?
+          File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar_markup) }
+        rescue StandardError => e
+          Jekyll.logger.error("           Sidebar: Unable to render sidebar for '#{filename}'.")
+          Jekyll.logger.debug("           Sidebar: #{e.message}. #{e.backtrace.inspect}")
+          nil
         end
 
-        if hash2.key? key
-          value = hash2[key]
+        sidebar_markup
+      end
 
-          hash_value = hash_value.merge(value) unless value.nil?
+      def find_containers(doc, filename)
+        sidebar_containers = doc.xpath('//*[@id="dx-sidebar-main-nav-ul"]')
+
+        unless sidebar_containers.any?
+          Jekyll.logger.debug("           Sidebar: No sidebar container found in #{filename}")
```

Comment:
Codacy found an issue: [Line is too long. [94/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)